### PR TITLE
Refactor Library Page to use form when filtering

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
# Description

On `ResearchLibraryPage`, the filtering logic is handled by a custom code. This changes the behaviour to use a Django `form` to get input from the user.

The *search* functionality was kept as it was.

Link to sample test page:
Related PRs/issues: #10392 

# Checklist

**Tests**
- [X] Is the code I'm adding covered by tests?

**Changes in Models:**
- ~~[ ] Did I update or add new fake data?~~
- ~~[ ] Did I squash my migration?~~
- ~~[ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/forms.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        "original_publication_date",
+        "year",
+        order="DESC",
+    )
+    year_options = [(date.year, date.year) for date in dates]
+    empty_option = (
+        "",
+        pgettext_lazy("Option in a list of years", "Any"),
+    )
+    return [empty_option] + year_options
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/pagemodels/research_hub/forms.py:38]: Moved all of these to this module (they don't depend on the Library Page) and changed the output to tuples (the expected format on the `choices` parameter

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py [hunk 2]
```diff
@@ -61,20 +62,20 @@ def get_context(self, request: "http.HttpRequest") -> "template.Context":
...
+            # This will result in all research being displayed.
+            filtered_author_ids: list[int] = []
+            filtered_topic_ids: list[int] = []
+            filtered_region_ids: list[int] = []
+            filtered_year: Optional[int] = None
+
+        filtered_author_ids = filter_form.cleaned_data["author"]
+        filtered_topic_ids = filter_form.cleaned_data["topic"]
+        filtered_region_ids = filter_form.cleaned_data["region"]
+        filtered_year = filter_form.cleaned_data["year"]
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/pagemodels/research_hub/library_page.py:78]: Not sure if these should be annotate as well, maybe they should

File: network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/research_filter_group.html
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/fragments/research_filter_group.html:1]: Not needed anymore, replaced by the filter fragment

File: network-api/networkapi/wagtailpages/templates/wagtailpages/research_library_page.html [hunk 3]
```diff
@@ -1,35 +1,31 @@
...
       that adds the `#filter` part to the URL. Submitting the form in that stage without the
       explicit URL would carry the anchor link forward. That would mean the view is scrolled to the
       filter section again upon page reload. This seems undesireable.
-    {% endcomment %}
-        <form action="{% pageurl page %}" method="get" accept-charset="utf-8" id="search-form">
+        {% endcomment %}
+        <form action="{% pageurl page %}"
+              method="get"
+              accept-charset="utf-8"
+              id="search-form">
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/research_library_page.html:20]: I decided to separate the search form from the filter form. The rendering of the filter form gets very messy if adding the search field, as well as the html here. The functionality should be the same, though.

File: network-api/networkapi/wagtailpages/tests/research_hub/test_forms.py [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+        region_2 = taxonomies_factory.ResearchRegionFactory()
+
+        region_options = forms._get_region_options()
+        region_option_values = [i for i, _ in region_options]
+
+        self.assertEqual(len(region_option_values), 2)
+        self.assertIn(region_1.id, region_option_values)
+        self.assertIn(region_2.id, region_option_values)
+
+    def test_region_in_options_matches_active_locale(self):
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/tests/research_hub/test_forms.py:158]: This test is broken right now depending on a fix for #10395.

File: network-api/networkapi/wagtailpages/tests/research_hub/test_forms.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+        topic_2 = taxonomies_factory.ResearchTopicFactory()
+
+        topic_options = forms._get_topic_options()
+        topic_option_values = [i for i, _ in topic_options]
+
+        self.assertEqual(len(topic_option_values), 2)
+        self.assertIn(topic_1.id, topic_option_values)
+        self.assertIn(topic_2.id, topic_option_values)
+
+    def test_topic_in_options_matches_active_locale(self):
```
Comment (jhonatan-lopes) [inline network-api/networkapi/wagtailpages/tests/research_hub/test_forms.py:107]: This test is broken right now depending on a fix for #10395.

File: network-api/networkapi/wagtailpages/pagemodels/research_hub/forms.py (hunk 1 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/research_hub/forms.py:38]: Thanks Jhonatan! This makes sense to me, 👍

**Review by danielfmiranda — APPROVED** [review]
Thanks @jhonatan-lopes! 

Looks good to me, LGTM!

**Pull request merged**